mods/pvp/grenades/init.lua

```diff
@@ -0,0 +1,140 @@
+		textures = {def.image},
+		collisionbox = {-0.2, -0.2, -0.2, 0.2, 0.15, 0.2},
+		pointable = false,
+		static_save = false,
+		particle = 0,
+		on_step = function(self, dtime)
+			local obj = self.object
+			local vel = obj:get_velocity()
+			local pos = obj:get_pos()
+
+			self.timer = self.timer + dtime
+
+			if not self.last_vel then
+				self.last_vel = vel
+			end
+
+			-- Collision Check
+
+			if not vector.equals(self.last_vel, vel) and vector.distance(self.last_vel, vel) > 4 then
+				if math.abs(self.last_vel.z - vel.z) > 6 then
+					self.last_vel.z = self.last_vel.z * -0.5
+				end
+
+				if math.abs(self.last_vel.x - vel.x) > 6 then
+					self.last_vel.x = self.last_vel.x * -0.5
+				end
+
+				if math.abs(self.last_vel.y - vel.y) > 6 then
+					self.last_vel.y = self.last_vel.y * -0.3
+				end
+
+				obj:set_velocity(self.last_vel)
+				vel = obj:get_velocity()
+			end
+
+			if self.sliding == 0 and vel.y == 0 then
+				self.sliding = 1
+			elseif self.sliding == 1 and vel.y ~= 0 then
+				self.sliding = 0
+			end
+
+			-- Can't use set_acceleration() because the grenade will shoot backwards once the velocity reaches 0
+
+			vel.x = vel.x / (1.036 + (self.sliding * 0.2))
+			vel.z = vel.z / (1.036 + (self.sliding * 0.2))
+
+			obj:set_velocity(vel)
+			self.last_vel = vel
+
+			-- Grenade Particles
+
+			if def.particle and self.particle >= 4 then
+				self.particle = 0
+
+				minetest.add_particle({
+					pos = obj:get_pos(),
+					velocity = vector.divide(vel, 2),
+					acceleration = vector.divide(obj:get_acceleration(), -5),
+					expirationtime = def.particle.life,
+					size = def.particle.size,
+					collisiondetection = false,
+					collision_removal = false,
+					vertical = false,
+					texture = def.particle.image,
+					glow = def.particle.glow
+				})
+			elseif def.particle and self.particle < def.particle.interval then
+				self.particle = self.particle + 1
+			end
+
+			-- Explode when clock is up
+
+			if self.timer > def.clock or not self.thrower_name then
+				if self.thrower_name then
+					minetest.log("[Grenades] A grenade thrown by " .. self.thrower_name ..
+					" explodes at " .. minetest.pos_to_string(vector.round(pos)))
+					def.on_explode(pos, self.thrower_name)
+				end
+
+				obj:remove()
+			end
+		end
+	}
+
+	minetest.register_entity(name, grenade_entity)
+
+	local newdef = {}
+
+	newdef.description = def.description
+	newdef.stack_max = 1
+	newdef.range = 0
+	newdef.inventory_image = def.image
+	newdef.on_use = function(itemstack, user, pointed_thing)
+		local player_name = user:get_player_name()
+
+		if pointed_thing.type ~= "node" then
+			local grenade = throw_grenade(name, user)
+			grenade.timer = 0
+			grenade.thrower_name = player_name
+
+			if not minetest.settings:get_bool("creative_mode") then
+				itemstack = ""
+			end
+		end
+
+		return itemstack
+	end
+
+	minetest.register_craftitem(name, newdef)
+end
+
+dofile(minetest.get_modpath("grenades").."/grenades.lua")
```

Comment:
Did you actually remove the spaces around the `..`? o_O

```suggestion
dofile(minetest.get_modpath("grenades") .. "/grenades.lua")
```

Reply:
I have no idea

Reply:
Fixed now